### PR TITLE
Add Go sample app

### PR DESCRIPTION
**Pull request by damemi**
Fixes https://github.com/GoogleCloudPlatform/opentelemetry-operator-sample/issues/30

### CODE REVIEW

**Comment by damemi** [comment]
Should work once https://github.com/open-telemetry/opentelemetry-operator/pull/1915 is released in the operator

File: sample-apps/go/app/go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/GoogleCloudPlatform/opentelemetry-operator-sample/sample-apps/go/app
```
Comment (dashpole) [inline sample-apps/go/app/go.mod:1]: nit: maybe move this to sample-apps/go/client instead?  Makes it clearer what it does, IMO

**Review by dashpole — APPROVED** [review]

File: sample-apps/go/app/go.mod (hunk 1 above)
Comment (damemi) [reply sample-apps/go/app/go.mod:1]: I did consider that, but the other samples all use `app` for their main code (they are built into the same binary). So, the shared commands to replace things like `%GCP_PROJECT` in k8s manifests expect `app` and `app.yaml`. For consistency when reading through this sample I decided to leave it as app here too

Comment (dashpole) [reply sample-apps/go/app/go.mod:1]: sg

**Comment by damemi** [comment]
Merging this. it still may not work until the new upstream operator release is published but that should be any day

**Pull request merged**